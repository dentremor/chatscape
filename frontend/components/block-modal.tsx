import { Modal, ModalVariant, Button } from "@patternfly/react-core";

export const BlockModal = ({
  name,
  modalOpen,
  closeModal,
  blockContact,
}: {
  name: string;
  modalOpen: boolean;
  closeModal: () => void;
  blockContact: () => void;
}) => (
  <Modal
    bodyAriaLabel="Block modal"
    tabIndex={0}
    variant={ModalVariant.small}
    title={`Block ${name}`}
    isOpen={modalOpen}
    onClose={closeModal}
    actions={[
      <Button
        key="confirm"
        variant="danger"
        onClick={() => {
          blockContact();
          closeModal();
        }}
      >
        Block
      </Button>,
      <Button key="cancel" variant="link" onClick={closeModal}>
        Cancel
      </Button>,
    ]}
  >
    Are you sure you want to block {name}? You will have to manually add them as
    a contact if you want to contact them again.
  </Modal>
);
