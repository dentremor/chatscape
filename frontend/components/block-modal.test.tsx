import { render } from "@testing-library/react";
import { BlockModal } from "./block-modal";

describe("BlockModal", () => {
  it("renders in the closed state", () => {
    expect(
      render(
        <BlockModal
          name=""
          modalOpen={false}
          closeModal={jest.fn()}
          blockContact={jest.fn()}
        />
      )
    ).toMatchSnapshot();
  });

  it("renders in the opened state", () => {
    expect(
      render(
        <BlockModal
          name=""
          modalOpen={true}
          closeModal={jest.fn()}
          blockContact={jest.fn()}
        />
      )
    ).toMatchSnapshot();
  });
});
