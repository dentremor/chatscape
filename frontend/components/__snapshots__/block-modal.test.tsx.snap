// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockModal renders in the closed state 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BlockModal renders in the opened state 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="pf-c-backdrop__open"
  >
    <div
      aria-hidden="true"
    />
    <div>
      <div
        class="pf-c-backdrop"
      >
        <div
          class="pf-l-bullseye"
        >
          <div
            aria-describedby="pf-modal-part-3"
            aria-labelledby="pf-modal-part-2"
            aria-modal="true"
            class="pf-c-modal-box pf-m-sm"
            data-ouia-component-id="OUIA-Generated-Modal-small-2"
            data-ouia-component-type="PF4/ModalContent"
            data-ouia-safe="true"
            id="pf-modal-part-1"
            role="dialog"
          >
            <button
              aria-disabled="false"
              aria-label="Close"
              class="pf-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Modal-small-2-ModalBoxCloseButton"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 352 512"
                width="1em"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                />
              </svg>
            </button>
            <header
              class="pf-c-modal-box__header"
            >
              <h1
                class="pf-c-modal-box__title"
                id="pf-modal-part-2"
              >
                <span
                  class="pf-c-modal-box__title-text"
                >
                  Block 
                </span>
              </h1>
            </header>
            <div
              aria-label="Block modal"
              class="pf-c-modal-box__body"
              id="pf-modal-part-3"
              role="region"
              tabindex="0"
            >
              Are you sure you want to block 
              ? You will have to manually add them as a contact if you want to contact them again.
            </div>
            <footer
              class="pf-c-modal-box__footer"
            >
              <button
                aria-disabled="false"
                class="pf-c-button pf-m-danger"
                data-ouia-component-id="OUIA-Generated-Button-danger-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                Block
              </button>
              <button
                aria-disabled="false"
                class="pf-c-button pf-m-link"
                data-ouia-component-id="OUIA-Generated-Button-link-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                Cancel
              </button>
            </footer>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
